lib/madmin/fields/belongs_to.rb

```diff
@@ -6,7 +6,9 @@ def options_for_select(record)
 
         klass = association.klass
         klass.all.map do |r|
-          ["#{klass.name} ##{r.id}", r.id]
+          name = Madmin.resource_for(r).display_name(r)
+
+          [name, r.id]
```

Comment:
Good call here!

Let's try to cache the resource so we don't have to look it up every iteration.

This might work so:
```ruby
      def options_for_select(record)
        association = record.class.reflect_on_association(attribute_name)
        klass = association.klass
        resource = nil
        klass.all.map do |r|
          resource ||= Madmin.resource_for(r)
          [resource.display_name(r), r.id]
        end
      end
```